app/actions/ForumActions.ts

```diff
@@ -0,0 +1,144 @@
+import { forumSchema, threadSchema } from 'app/reducers';
+import { Forum, Thread } from './ActionTypes';
+import callAPI from './callAPI';
+import type { ID } from 'app/models';
+import type {
+  CreateForum,
+  CreateThread,
+  DetailedForum,
+  DetailedThread,
+  UpdateForum,
+  UpdateThread,
+} from 'app/store/models/Forum';
+
+export function fetchForums() {
+  return callAPI({
```

Comment:
```suggestion
  return callAPI<DetailedForum[]>({
```
Or `PublicForum`? Too tired to check